experimenter/pipeline_builder.py

```diff
@@ -0,0 +1,55 @@
+from d3m import index as d3m_index
+from d3m.metadata.pipeline import Pipeline
+from d3m.metadata.pipeline import PrimitiveStep
+from d3m.metadata.base import ArgumentType
+
+def create_pipeline_step(
+    step_counter: int,
```

Comment:
The only thing that worries me about this refactor is if D3M changes these primitives to take extra parameters.  I thought the previous code would look up the number of arguments dynamically, while this seems to assume it only takes `inputs`.  But they may never change that, so idk.  It'd be a good question for @bjschoenfeld

Reply:
Like you state down below Orion, these helpers can be used for situations that fit, but don't need to be used everywhere. We are still preserving the dynamic reading in of arguments.

Reply:
As @epeters3 said, this is a help for when the primitive only needs one argument, `inputs`. D3M is becoming more stable, so I don't expect dramatic changes moving forward. I like adding helper functions as they become useful.